### PR TITLE
Stop passing a JupyterHub API token for use by health check

### LINKED ISSUE #1024: Generate token for Health Check service in the z2jh helm chart instead of in the deployer
### Background and proposal

**Context**

We currently have two functions in the deployer that generates helm values on-the-fly that get parsed to the `helm upgrade` command:

- [`get_generated_config`](https://github.com/2i2c-org/infrastructure/blob/aa580cb0c28f4ef6599b6ba1a1a3fae9b1ac33f2/deployer/hub.py#L348-L466)
- [`apply_hub_helm_chart_fixes`](https://github.com/2i2c-org/infrastructure/blob/aa580cb0c28f4ef6599b6ba1a1a3fae9b1ac33f2/deployer/hub.py#L479-L518)

In `apply_hub_helm_chart_fixes`, [an API token is generated with admin privileges](https://github.com/2i2c-org/infrastructure/blob/aa580cb0c28f4ef6599b6ba1a1a3fae9b1ac33f2/deployer/hub.py#L492-L499). The deployer script later [uses this API token in the testing phase](https://github.com/2i2c-org/infrastructure/blob/aa580cb0c28f4ef6599b6ba1a1a3fae9b1ac33f2/deployer/hub.py#L600-L627). It uses the token to create a singleuser server on which some notebooks are executed top-to-bottom to ensure the hub is working at a basic level.

**Problem statement:** The more config that is not explicitly stated in a helm chart values file, and is instead generated on-the-fly in the deployer's "black box", the more difficult our deployment process becomes for our engineers to understand, and the more difficult a deployment is to replicate for a customer choosing to leave 2i2c and take their hub with them.

**Proposed solution**

The z2jh helm chart has the ability to automatically generate tokens (if they do not already exist) for registered services and store them as kubernetes secrets. The deployer, using service account credentials, will have permission to read kubernetes secrets in order to retrieve the token and execute its tests as normal. This would reduce the amount of on-the-fly-generated config produced by the deployer's "black box" and explicitly define more config in the form of helm chart values, or direct configuration in the basehub helm chart. Hence improving the comprehensibility of our deployment process and simplifying the amount of tasks the deployer carries out.

### Implementation guide and constraints

_No response_

### Updates and ongoing work

_No response_

### PR DESCRIPTION
**Pull request by consideRatio**
Recently the deployer script was made to acquire a JupyterHub API token
by inspecting the `hub` k8s Secret created by the JupyterHub Helm chart
for when it was to interact with the JupyterHub API to make health
checks.

As this API token in modern versions of the JupyterHub Helm chart will
be generated automatically for `hub.services` registered JupyterHub
services, or reused in case it already is defined in the k8s Secret, we
are able to stop defining this token explicitly over and over.

In other words:

- For new installations: the API token will be generated by the
  JupyterHub Helm chart instead
- For existing installations: the API token already set in the k8s
  Secret `hub` will be re-used by helm 3's `lookup` function as used by
  the JupyterHub Helm chart.

---

- Together with #1185, this closes #1024.

### CODE REVIEW

File: deployer/hub.py [hunk 1]
```diff
@@ -159,22 +159,22 @@ def apply_hub_helm_chart_fixes(self, generated_config, secret_key):
...
+        #        function. Currently we only generate a JupyterHub API token for
+        #        dask-gateway to use, but as that is resolved we can cleanup
+        #        this entire function.
+        #
         if hub_helm_chart != "basehub":
             generated_config = {"basehub": generated_config}
 
-        # FIXME: This section can be fixed upon resolution of:
-        # https://github.com/dask/dask-gateway/issues/473
+        # FIXME: This section can be removed upon resolution of, where we would
```
Comment (sgibson91) [inline deployer/hub.py:170]:
```suggestion
        # FIXME: This section can be removed upon resolution of the below linked issue, where we would
```

I think?

**Review by sgibson91 — APPROVED** [review]
LGTM!
